lib/diagnostic.rb

```diff
@@ -43,6 +45,9 @@ def transform(words)
   # replacing the trailing newline character with a
   # space and return the result.
   def read_file(filename)
-    # Your code goes here
+    File.open('filename').each do |line|
+      line.sub! newline, ' '
```

Comment:
You are missing a step. When you open the file, you a file to iterate through, you have to get down to a line

```ruby
File.open('filename') do |file|
   file.each do |line|
    line.chomp(' ')
   end
end
```